Testscripts/Linux/Linux-Kernel-Self-Tests.sh

```diff
@@ -0,0 +1,322 @@
+	done
+	TOTAL_TARGETS=$(echo $target_item | sed -e 's/^[ ]*//g')
+}
+
+# TOTAL_TARGETS is assiged a items string list of the intersection of $1 and $2
+function check_targets() {
+	target_item=""
+	for _tar in $2; do
+		flag=0
+		for _tar_item in $1; do
+			if [ "$_tar"x  == "$_tar_item"x ]; then
+				flag=1
+				break
+			fi
+		done
+		if [ $flag -eq 1 ]; then
+			target_item=$target_item" $_tar"
+		fi
+	done
+	TOTAL_TARGETS=$(echo $target_item | sed -e 's/^[ ]*//g')
+}
+
+function build_and_run_lks() {
+	cd $LKS_SRCDIR
+
+	if [ -f $MAKEFILE ]; then
+		targets=$(cat $MAKEFILE | grep TARGETS | awk -F '=' '{print $2}' | sort -u | grep -v "^$")
+		TOTAL_TARGETS="$targets"
+
+		if [ "$TARGETS" != "" ]; then 
+			check_targets "$targets" "$TARGETS"
+		fi
+		
+		if [[ $DISTRO =~ "centos" || $DISTRO =~ "redhat" ]]; then
+			LogMsg "skip targets ($SKIP_TARGETS_FOR_CENTOS) for Centos or Redhat"
+			skip_targets "$TOTAL_TARGETS" "$SKIP_TARGETS_FOR_CENTOS"
+		fi
+
+		if [[ $DISTRO =~ "ubuntu" ]]; then
+			version_id=$(cat /etc/os-release | grep VERSION_ID | sed 's/VERSION_ID=//g' | sed 's/\"//g' | awk -F '.' '{print $1}')
+			if [ $[$version_id] -le 16 ]; then
+				LogMsg "skip targets ($SKIP_TARGETS_FOR_UBUNTU16) for Ubuntu 16 or lower version"
+				skip_targets "$TOTAL_TARGETS" "$SKIP_TARGETS_FOR_UBUNTU16"
+			fi
+		fi
+
+		if [ "$SKIP_TARGETS" != "" ]; then
+			LogMsg "skip targets $SKIP_TARGETS for these targets may cause system hanging"
+			skip_targets "$TOTAL_TARGETS" "$SKIP_TARGETS"
+		fi
+
+		LogMsg "total targets: $TOTAL_TARGETS"
+		if [ $SUMMARY -eq 1 ]; then
+			make -C tools/testing/selftests summary=1 TARGETS="$TOTAL_TARGETS" run_tests >> $LKS_OUTPUT 2>&1
+		else
+			make -C tools/testing/selftests TARGETS="$TOTAL_TARGETS" run_tests >> $LKS_OUTPUT 2>&1
+		fi
+	else
+		LogMsg "Can't find lks Makefile"
+		SetTestStateAborted
+		exit 0
+	fi
+}
+
+#######################################################################
+#
+# Main script body
+#
+#######################################################################
+# Source utils.sh
+. utils.sh || {
+    echo "Error: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+HOMEDIR=$(pwd)
+LKS_RESULTS="$HOMEDIR/lks-results.log"
+LKS_OUTPUT="$HOMEDIR/lks-output.log"
+BUILDING_LOG="$HOMEDIR/lks-building.log"
+VM_LOG="$HOMEDIR/VM_properties.csv"
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Clear the old log
+rm -f $LKS_RESULTS $LKS_OUTPUT
+
+LogMsg "Installing dependencies"
+install_dependencies
+
+# define regular stable releases in order to avoid unstable builds
+# git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
+# 'LKS_VERSION_GIT_TAG' is passed from Test Definition in .\XML\TestCases\CommunityTests.xml.
+# 'LKS_VERSION_GIT_TAG' default value is defined in .\XML\Other\ReplaceableTestParameters.xml
+LogMsg "Download LKS source code"
+if [[ "$LKS_VERSION_GIT_TAG" != "" ]]; then
+	LogMsg "kernel source git: $stable_kernel_src"
```

Comment:
Indentation? https://github.com/LIS/LISAv2/blob/master/Documents/How-to-use.md I think we are using 4-space tab.

Reply:
@lubaihua33  I see you still have different indentation in visual. Can you check again?